pyanaconda/packaging/__init__.py

```diff
@@ -636,29 +636,11 @@ def writeStorageLate(self):
            every payload except for dnf.  Payloads should only implement one of
            these methods by overriding the unneeded one with a pass.
         """
-        if not flags.dirInstall:
-            if iutil.getSysroot() != iutil.getTargetPhysicalRoot():
-                setSysroot(iutil.getTargetPhysicalRoot(), iutil.getSysroot())
-
-                # Now that we have the FS layout in the target, umount
-                # things that were in the legacy sysroot, and put them in
-                # the target root, except for the physical /.  First,
-                # unmount all target filesystems.
-                self.storage.umountFilesystems()
-
-                # Explicitly mount the root on the physical sysroot
-                rootmnt = self.storage.mountpoints.get('/')
-                rootmnt.setup()
-                rootmnt.format.setup(options=rootmnt.format.options, chroot=iutil.getTargetPhysicalRoot())
-
-                self.prepareMountTargets(self.storage)
-
-                # Everything else goes in the target root, including /boot
-                # since the bootloader code will expect to find /boot
-                # inside the chroot.
-                self.storage.mountFilesystems(skipRoot=True)
-
-            self.storage.write()
+        if iutil.getSysroot() != iutil.getTargetPhysicalRoot():
+            setSysroot(iutil.getTargetPhysicalRoot(), iutil.getSysroot())
+            self.prepareMountTargets(self.storage)
+            if not flags.dirInstall:
+                self.storage.write()
```

Comment:
I don't know if we should care but for
flags.dirInstall == False, iutil.getSysroot() == iutil.getTargetPhysicalRoot() we do not call self.storage.write() anymore with the patch.